[xenyeo] iP

xenyeo opened the pull request:
# Kaji
>"Your mind is for having ideas, not holding them." - David Allen ([source](https://dansilvestre.com/productivity-quotes/))

Kaji frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ _SUPER_ FAST to use

All you need to do is,
1. download it from [here](https://nus-cs2103-ay2425s2.github.io/website/schedule/week4/project.html)
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉 

And it is **FREE**!

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

---

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

xenyeo committed:
Renamed chatbot and implemented initial skeleton (greeting)

xenyeo committed:
Improved skeleton of chatbot. (Added support for echo)

xenyeo committed:
Improved skeletal version of UI

xenyeo committed:
Added list to store and display text when requested.

xenyeo committed:
Update storage type to Task

xenyeo committed:
Fixed bug regarding list command

xenyeo committed:
Enhanced echo UI

xenyeo committed:
Indentation and separator can now be extracted as string

xenyeo committed:
Added class to represent Task

xenyeo committed:
Added ability to mark tasks as done

xenyeo committed:
Added cases for new task types

xenyeo committed:
Added skeleton for 3 task types

xenyeo committed:
Added support for ToDos, Events, Deadlines

xenyeo committed:
Added exception handling to all possible errors in current version

xenyeo committed:
Added exception handling to all possible errors in current version

xenyeo committed:
Changed array implementation to ArrayList<Task>

xenyeo committed:
Added exception handling for duplicate marking and unmarking

xenyeo committed:
Added support for delete

xenyeo committed:
Tested I/O with level-4 as reference

xenyeo committed:
Add Storage.java

xenyeo committed:
Add boilerplate code to Storage.java

xenyeo committed:
Add saveTasks method to Storage.java

xenyeo committed:
Add loadTasks method to Storage.java

xenyeo committed:
Fix bugs caused by Storage.java

xenyeo committed:
Add exception handling for invalid file format in kaji.txt

xenyeo committed:
Add convertDateTime method to TaskList.java

xenyeo committed:
Update format for deadline tasks

xenyeo committed:
Add isSameDate method to Event.java

xenyeo committed:
Update format for event tasks

xenyeo committed:
Add exception handling for invalid date format

xenyeo committed:
Add fileLoadException.java

xenyeo committed:
Update loadTasks method to load

xenyeo committed:
Update main skeleton for Kaji.java

xenyeo committed:
Update kaji.java OOP format

xenyeo committed:
Update Ui.java OOP format

xenyeo committed:
Update storage.java OOP format

xenyeo committed:
Update TaskList.java OOP format

xenyeo committed:
Update Parser.java to OOP format

xenyeo committed:
Update KajiException.java

xenyeo committed:
Update task child classes to OOP format

xenyeo committed:
Update command child classes to OOP format

xenyeo committed:
Update date and time in Deadline.java

xenyeo committed:
Organize related files into kaji.command package

xenyeo committed:
Organize related files into kaji.task package

xenyeo committed:
Merge branch 'add-gradle-support'

* add-gradle-support:
  build.gradle: Prevent generating a second JAR file
  Bump gradle and lib version
  Add Gradle support

xenyeo committed:
Add ToDoTest.java for JUnit tests

xenyeo committed:
Add ParserTest.java for JUnit tests

xenyeo committed:
Update ToDoTest.java for JUnit tests

xenyeo committed:
Update build.gradle

xenyeo committed:
Update MANIFEST.MF

xenyeo committed:
Add java doc

xenyeo committed:
Tweaked codes to comply with coding standard

xenyeo committed:
Merge branch 'branch-A-CodingStandard'

* branch-A-CodingStandard:
  Tweaked codes to comply with coding standard

# Conflicts:
#	src/main/java/kaji/command/AddCommand.java

xenyeo committed:
Add FindCommand.java and its relevant codes

xenyeo committed:
Fix bugs introduced by Task.java

xenyeo committed:
Update kaji.java

xenyeo committed:
Update Ui.java

kennethsoh commented on src/main/java/kaji/Kaji.java:
```diff
@@ -0,0 +1,47 @@
+        storage = new Storage(filePath);
+        try {
+            tasks = new TaskList(storage.load());
+        } catch (KajiException e) {
```
I think that the naming "KajiException" is not useful in helping others understand exactly what exception is being caught.

kennethsoh commented on src/main/java/kaji/Storage.java:
```diff
@@ -0,0 +1,88 @@
+import kaji.task.Event;
+import kaji.task.Task;
+
+import java.io.*;
```
I think imported classes should be listed explicitly.

kennethsoh commented on src/main/java/kaji/Parser.java:
```diff
@@ -0,0 +1,151 @@
+package kaji;
+
+import kaji.command.*;
```
I think imported classes should be listed explicitly.

kennethsoh commented on src/main/java/kaji/TaskList.java:
```diff
@@ -0,0 +1,126 @@
+     * @param taskId the ID of the task to be deleted
+     */
+    public void deleteTask(int taskId, Ui ui) throws KajiException {
+            if (taskId > taskList.size() || taskId == 0) {
```
Is the indentation (4 spaces, not tabs) here correct? 🤔

kennethsoh commented on src/main/java/kaji/Ui.java:
```diff
@@ -0,0 +1,177 @@
+        Scanner sc = new Scanner(System.in);
+        return sc.nextLine();
+    }
+}
```
This file LTGM!

kennethsoh commented on src/test/java/kaji/ParserTest.java:
```diff
@@ -0,0 +1,131 @@
+
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
```
I think imported classes should be listed explicitly.

kennethsoh reviewed (commented):
Overall LTGM, except for some minor issues 👍🏻

xenyeo committed:
Add checkstyle related files

xenyeo committed:
Update files to adhere to coding standards

xenyeo committed:
Add GUI to chatbot using JavaFX

Codekrodile commented on src/main/java/kaji/DialogBox.java:
```diff
@@ -0,0 +1,60 @@
+import javafx.scene.control.Label;
+import javafx.scene.image.Image;
+import javafx.scene.image.ImageView;
+import javafx.scene.layout.HBox;
```
i like how the imports are lexicographically increasing order

Codekrodile commented on src/main/java/kaji/task/Deadline.java:
```diff
@@ -0,0 +1,52 @@
+     * Returns a string representation of the kaji.task.Deadline object.
+     *
+     * @return A string representation of the kaji.task.Deadline object.
+     */
```
consider making the Javadoc clearer by specifying what the output format is

ie "TaskDescription (by: MMM dd yyy, HH:mm")

Codekrodile commented on src/main/java/kaji/Kaji.java:
```diff
@@ -0,0 +1,56 @@
+
+    public Kaji() {
+        this.ui = new Ui();
+        this.storage = new Storage(FILE_PATH);
```
consider adding blank lines between logical sections to improve readability

Codekrodile commented on src/main/java/kaji/Kaji.java:
```diff
@@ -0,0 +1,56 @@
+            Command c = Parser.parse(input);
+            return c.execute(tasks, ui, storage);
+        } catch (KajiException e) {
+            return e.getMessage();
```
consider logging the error before this return statement, for example

System.err.println("Error processing input: " + e.getMessage());

Codekrodile commented on src/main/java/kaji/Kaji.java:
```diff
@@ -0,0 +1,56 @@
+     *
+     * @throws KajiException if an error occurs while loading the tasks
+     */
+    private void loadFile() {
```
consider a more descriptive function name, like loadTasksFromFile()

xenyeo committed:
Add assertions for system level validations.

The first assertion is to ensure that a file path exists for data storage.
The second assertion is to ensure that the data parsing from storage is of the right format.

xenyeo committed:
Refactor code to improve code quality

Update method signatures to remove unused parameters.
Update javadoc for understanding.
Replace nested conditions with guard clauses.

xenyeo committed:
Merge pull request #2 from xenyeo/branch-A-Assertions

Add assertions for system level validations.

xenyeo committed:
Merge pull request #3 from xenyeo/master

Sync master branch with coding standard branch

xenyeo committed:
Merge pull request #4 from xenyeo/master

Sync master branch to code quality branch

xenyeo committed:
Merge pull request #5 from xenyeo/branch-A-CodingStandard

Merge pull request #3 from xenyeo/master

xenyeo committed:
Add tagging to tasks

xenyeo committed:
Change profile picture

xenyeo committed:
Update phrases used as reply

xenyeo committed:
Fix bug issue pertaining to closing app

xenyeo committed:
Fix bug issues pertaining user input

xenyeo committed:
Add more details to user error response

xenyeo committed:
Fix bugs pertaining to adding and deleting of tags

xenyeo committed:
Add title bar to GUI

xenyeo committed:
Add anchor to TextField, Send Button, and ScrollPane

xenyeo committed:
Add limit to window size

xenyeo committed:
Add style to MainWindow.fxml

xenyeo committed:
Add style to DialogBox.fxml

xenyeo committed:
Update dialog box margin

xenyeo committed:
Update README.md

xenyeo committed:
Add Ui.png

xenyeo committed:
Update README.md

xenyeo committed:
Update README.md

xenyeo committed:
Update README.md
